buildrunner/steprunner/__init__.py

```diff
@@ -49,7 +50,7 @@ def __init__(
         self,  # pylint: disable=too-many-arguments
         build_runner,
         step_name,
-        step_config,
+        step: Step,
```

Comment:
Should we also add typing for the other arguments?

Reply:
I wasn't sure on all of the types, so I tried to limit it to what I was doing at the moment for the most part. Yes, they should be added at some point.

Reply:
Understandable. I agree it doesn't have to be done in this this PR.